tests/external_executor/external_executor.py

```diff
@@ -261,12 +270,21 @@ def test_kv(network, args):
 def test_simple_executor(network, args):
     primary, _ = network.find_primary()
 
-    credentials = register_new_executor(primary, network)
+    supported_endpoints = [
+        ("GET", "/article_description/Earth"),
+        ("POST", "/update_cache/Earth"),
+    ]
+
+    credentials = register_new_executor(
+        primary, network, supported_endpoints=supported_endpoints
+    )
 
     with executor_thread(WikiCacherExecutor(primary, credentials)):
         with primary.client() as c:
             r = c.post("/not/a/real/endpoint")
-            assert r.status_code == http.HTTPStatus.NOT_FOUND
+            body = r.body.json()
+            assert r.status_code == http.HTTPStatus.INTERNAL_SERVER_ERROR
```

Comment:
I don't think we should lose this 404, and convert it to a 500. I guess that's coming from the `logic_error` thrown in `validate_supported_endpoints`?

If we instead returned a `std::optional` from that, we could treat nullopt (=> no active executor can handle this request) into a 404.

Reply:
500 is being thrown by the test infra (as it encounters the `logic_error`).

Sure, we can return probably a nullopt and convert it in the test, but I'm not sure how that will translate to Executors? It seems better to throw 500 with a clear error message like it is now?

Reply:
We can still return a descriptive error in a 404. Since this is the error that will be returned when users mis-spell a uri, I think it should definitely be a 404 rather than a 500? 500s should generally indicate something that's definitely not the caller's fault, where some internal invariant is broken and we have nothing more specific to report - I don't think that's the case here.